internal/authorization/middlewares.go

```diff
@@ -132,7 +125,19 @@ func (mdw *Middleware) Authorize() func(http.Handler) http.Handler {
 
 				// if we got here then `principal` must be != nil
 				principal := authentication.PrincipalFromContext(r.Context())
-				isAdmin := mdw.isAdmin(principal)
+				if principal == nil {
+					// should never happen if authentication is configured correctly
+					mdw.logger.Fatalf("principal not available in context, cannot proceed with authorization")
```

Comment:
we cannot panic when a request is in flight, we can return 500 though and log hard